src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -2144,31 +2166,56 @@ internal void AddMemberLookupSymbolsInfo(LookupSymbolsInfo result, NamespaceOrTy
             }
         }
 
-        private void AddMemberLookupSymbolsInfoInType(LookupSymbolsInfo result, TypeSymbol type, LookupOptions options, Binder originalBinder)
+#nullable enable
+        /// <param name="accessThroughType">When collecting symbols from an extension type, we include the members from the extended type,
+        /// but those will be accessed through the extension type</param>
+        private void AddMemberLookupSymbolsInfoInType(LookupSymbolsInfo result, TypeSymbol type, LookupOptions options, Binder originalBinder, TypeSymbol? accessThroughType = null)
         {
+            Debug.Assert(accessThroughType is null || accessThroughType.IsExtension);
+
             switch (type.TypeKind)
             {
                 case TypeKind.TypeParameter:
+                    // PROTOTYPE(static) need to confirm what we want for extension members on type parameters or values of type parameters
+                    Debug.Assert(accessThroughType is null);
                     this.AddMemberLookupSymbolsInfoInTypeParameter(result, (TypeParameterSymbol)type, options, originalBinder);
                     break;
 
                 case TypeKind.Interface:
-                    this.AddMemberLookupSymbolsInfoInInterface(result, type, options, originalBinder, type);
+                    accessThroughType ??= type;
```

Comment:
>accessThroughType ??= type; [](http://example.com/codeflow?start=20&length=27)

This looks suspicious. The `accessThroughType` parameter of `AddMemberLookupSymbolsInfoInInterface` is used for accessibility check. So, it is not clear to me why would we want to use an extension type for that.